docs/source/array-creation.rst

```diff
@@ -201,6 +201,42 @@ on your Dask array to turn in to a normal NumPy array:
    array([0, 1, 4, 9, 16, 25])
 
 
+Intermediate storage
+--------------------
+
+.. autosummary::
+   store
+
+In some cases, one may wish to store an intermediate value that will be used in
+later computations. This could be useful for a variety of cases like saving
+work before a flaky step, as a simple form of caching, a useful
+restore/stopping point in a long computation, inclusion of the intermediate
+value in later analysis, etc.
+
+This uses ``store`` or anything that builds off of ``store`` to provide this
+functionality. The intermediate storage operation can be done in one of two
+ways. Triggering a computation immediately, which behaves like ``persist``
+except the computed value is placed in the ``targets`` instead. Alternatively
+computation can be delayed, which merely includes the storage step in the Dask
+graph. Both options allow this intermediate value to be used in later
+computations and both place the value in the ``targets``, which it reads the
+result from later.
```

Comment:
This paragraph starts off with "This uses ``store`` ..."  What does "This" refer to?  The example below?  

I think that this section would be better if it described two ways of storing intermediate results, starting with `persist` and then using `store`, describing why one would choose one or the other.  As-is it reads more like a motivation as to why people should use `store.`

Reply:
Sure that makes sense. Happy to add that.